main.go

```diff
@@ -2,78 +2,237 @@ package main
+	DefaultClusterDirectory     string = "cluster"
+	DefaultShowTemplates        bool   = false
+	DefaultNoGit                bool   = false
+	DefaultNoSecure             bool   = false
+	DefaultTFTPRoot             string = "./tftproot"
+	DefaultYochuPath            string = "./yochu"
+	DefaultStaticHTMLPath       string = "./static_html"
+	DefaultFirstStageScript     string = "./templates/first_stage_script.sh"
+	DefaultLastStageCloudconfig string = "./templates/last_stage_cloudconfig.yaml"
+	DefaultDnsmasqTemplate      string = "./templates/dnsmasq_template.conf"
+	DefaultTemplateSnippets     string = "./template_snippets"
+	DefaultDNSMasq              string = "/usr/sbin/dnsmasq"
+	DefaultImagesCacheDir       string = "./images"
+	DefaultHTTPPort             int    = 4080
+	DefaultHTTPBindAddress      string = "0.0.0.0"
+	DefaultTLSCertFile          string = ""
+	DefaultTLSKeyFile           string = ""
+	DefaultEtcdQuorumSize       int    = 3
+)
+
+type MayuFlags struct {
+	debug   bool
+	version bool
+
+	configFile           string
+	clusterDir           string
+	showTemplates        bool
+	noGit                bool
+	noSecure             bool
+	tFTPRoot             string
+	yochuPath            string
+	staticHTMLPath       string
+	firstStageScript     string
+	lastStageCloudconfig string
+	templateSnippets     string
+	dnsmasq              string
+	dnsmasqTemplate      string
+	imagesCacheDir       string
+	httpPort             int
+	httpBindAddress      string
+	tlsCertFile          string
+	tlsKeyFile           string
+	etcdQuorumSize       int
+
+	filesystem fs.FileSystem // internal filesystem abstraction to enable testing of file operations.
+}
+
 var (
-	confFile      = flag.String("config", "/etc/mayu/config.yaml", "path to the configuration file")
-	clusterDir    = flag.String("cluster-directory", "cluster", "path cluster directory")
-	showTemplates = flag.Bool("show-templates", false, "show the templates and quit")
-	noGit         = flag.Bool("no-git", false, "disable git operations")
-	showVersion   = flag.Bool("version", false, "show the version of mayu")
-
-	conf      configuration
-	tempFiles = make(chan string, 4096)
+	globalFlags = MayuFlags{}
+
+	mainCmd = &cobra.Command{
+		Use:   "mayu",
+		Short: "Manage your bare metal machines",
+		Long:  "",
+		Run:   mainRun,
+	}
+
+	projectVersion string
+	projectBuild   string
 )
 
-func main() {
-	log.SetFlags(0)
-	log.SetPrefix("mayu: ")
-	flag.Set("logtostderr", "true")
-	flag.Parse()
+func init() {
+	// make sure Mayu logs to stderr
+	if err := flag.Lookup("logtostderr").Value.Set("true"); err != nil {
+		panic(err)
+	}
 
-	if *showVersion {
-		printVersion()
-		os.Exit(0)
+	// Map any flags registered in the standard "flag" package into the
+	// top-level mayu command (eg. log flags)
+	pf := mainCmd.PersistentFlags()
+	flag.VisitAll(func(f *flag.Flag) {
+		pf.AddFlag(pflag.PFlagFromGoFlag(f))
+	})
+
+	pf.BoolVarP(&globalFlags.debug, "debug", "d", false, "Print debug output")
```

Comment:
Why ? We're doing this in most of our repos. You could also see other projects... It's just a matter of consistency